spynnaker/pyNN/models/neuron/synapse_dynamics/abstract_synapse_dynamics.py

```diff
@@ -104,28 +104,28 @@ def get_provenance_data(self, pre_population_label, post_population_label):
         # pylint: disable=unused-argument
         return list()
 
-    def get_delay_maximum(self, connector, delays):
+    def get_delay_maximum(self, connector, delays, synapse_info):
         """ Get the maximum delay for the synapses
         """
-        return connector.get_delay_maximum(delays)
+        return connector.get_delay_maximum(delays, synapse_info)
 
     def get_delay_variance(self, connector, delays):
         """ Get the variance in delay for the synapses
         """
         # pylint: disable=too-many-arguments
         return connector.get_delay_variance(delays)
 
-    def get_weight_mean(self, connector, weights):
+    def get_weight_mean(self, connector, weights, synapse_info):
```

Comment:
Similarly to connectors, this can drop the weights as this comes from synapse_info.